clients/src/main/java/org/apache/kafka/clients/FetchSessionHandler.java

```diff
@@ -301,7 +299,9 @@ public FetchRequestData build() {
                 if (nextData != null) {
                     // We basically check if the new partition had the same topic ID. If not,
                     // we add it to the "replaced" set.
-                    if (!prevData.topicId.equals(nextData.topicId) && !prevData.topicId.equals(Uuid.ZERO_UUID)) {
+                    if (!prevData.topicId.equals(nextData.topicId)
+                            && !prevData.topicId.equals(Uuid.ZERO_UUID)
+                            && !nextData.topicId.equals(Uuid.ZERO_UUID)) {
```

Comment:
Is there a reason we do this? If the previous data had a topic ID and this one doesn't we should send a different fetch request version and the session will be closed.

Reply:
Without this, when a topic id is set back to "zero", the former topic id is added to the replaced set which is a bit unintuitive, I think. In the end, it does not matter too much because the version is downgraded in this case so the replaced set is ignored. I was debating if it worth handling this case explicitly here.

Reply:
I realize we may still want this as if the partition data is exactly the same, we will actually ignore the downgrade which is not good.

Reply:
I have a test where this happens. 😅